### PR TITLE
Fix in EDP because wasn't setting first QoS correctly [8703]

### PR DESCRIPTION
**Pull request by richiware**
For example, a user setting the UserData QoS, it is not be sent in the discovery.
I've added a regression test.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2195/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1984/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2189/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2198/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1987/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2192/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2200/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1989/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2194/)

File: test/blackbox/api/dds-pim/PubSubWriter.hpp [hunk 1]
```diff
@@ -311,10 +311,10 @@ class PubSubWriter
                 datawriter_ = publisher_->create_datawriter(topic_, datawriter_qos_, &listener_, status_mask_);
                 if (datawriter_ != nullptr)
                 {
-                    std::cout << "Created datawriter " << datawriter_->guid() << " for topic " <<
+                    datawriter_guid_ = datawriter_->guid();
+                    std::cout << "Created datawriter " << datawriter_guid_ << " for topic " <<
                         topic_name_ << std::endl;
                     initialized_ = datawriter_->is_enabled();
-                    return;
```
Comment (MiguelCompany) [inline test/blackbox/api/dds-pim/PubSubWriter.hpp:317]: Please restore this `return;`

File: test/blackbox/api/fastrtps_deprecated/PubSubWriter.hpp [hunk 2]
```diff
@@ -298,10 +300,10 @@ class PubSubWriter
 
             if (publisher_ != nullptr)
             {
-                std::cout << "Created publisher " << publisher_->getGuid() << " for topic " <<
+                publisher_guid_ = publisher_->getGuid();
+                std::cout << "Created publisher " << publisher_guid_ << " for topic " <<
                     publisher_attr_.topic.topicName << std::endl;
                 initialized_ = true;
-                return;
```
Comment (MiguelCompany) [inline test/blackbox/api/fastrtps_deprecated/PubSubWriter.hpp:304]: Please restore this `return;`

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2397/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2198/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2388/)

**Review by MiguelCompany — DISMISSED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2401/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2201/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2391/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2915/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/201/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2667/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2926/)

**Comment by MiguelCompany** [comment]
@IkerLuengo Please fix uncrustify

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**